src/membrane_proof.rs

```diff
@@ -87,52 +83,61 @@ fn mem_proof_server_url() -> String {
     }
 }
 
-pub fn get_hpos_config() -> Result<Config> {
-    let config_path = env::var("HPOS_CONFIG_PATH")?;
-    let config_json = fs::read(config_path)?;
-    let config: Config = serde_json::from_slice(&config_json)?;
-    Ok(config)
+/// Returns memproof from a file at MEM_PROOF_PATH
+/// If a file does not exist then function downloads existing mem-proof for given agent
+/// from HBS server and saves it to the file
+/// Returns error if no memproof obtained, because memproof is mandatory
+/// for core-app installation
+#[instrument(skip(admin), err)]
+pub async fn get_mem_proof(admin: Admin) -> Result<MembraneProof> {
+    if let Ok(m) = load_mem_proof_from_file() {
+        return Ok(m);
+    }
+
+    let mem_proof_str = download_memproof(admin).await?;
+    save_mem_proof_to_file(&mem_proof_str)?;
+
+    let mem_proof_bytes = base64::decode(mem_proof_str)?;
+    let mem_proof_serialized = Arc::new(SerializedBytes::from(UnsafeBytes::from(mem_proof_bytes)));
+    Ok(mem_proof_serialized)
+}
+
+/// Creates HashMap of memproofs for dnas based on happ id
+/// which is later consumed during happ installation
+/// Currently creates memproofs only for core-app
+/// otherwise returns empty HashMap
+/// Returns HashMap<dna_name, memproof_bytes>
+pub async fn crate_vec_for_happ(
```

Comment:
Typo in function name?

Reply:
Yeah, I should come up with better name

Reply:
addressed in [982b4e7](https://github.com/Holo-Host/hpos-configure-holochain/pull/143/commits/982b4e70a41e71124209de89c93ca8fef8311e58)